frontend/components/AddRemoveButton.tsx

```diff
@@ -1,12 +1,23 @@
 import React from 'react';
 import AddIcon from '@mui/icons-material/Add';
 import RemoveIcon from '@mui/icons-material/Remove';
-import { Button, SvgIcon, Typography } from '@mui/material';
+import { Button, SvgIconProps, Typography } from '@mui/material';
 
-export default function AddRemoveButton (props) {
+interface AddRemoveButtonProps {
```

Comment:
Have you considered extending Button like described here? https://mui.com/material-ui/guides/typescript/#complications-with-the-component-prop

Is there an reason to do a custom interface? I'm just learning here so just tying to find best practice for this.

Reply:
Maybe there's something I can simplify around `ButtonProps`, I will take a look. We could have some of these interfaces extend from `mui.ButtonProps`, but we are setting some of these values internally based on the `navEnabled` flag.

Extending `Button`, I'm not sure about. I think I'm kind of following the composition approach they describe, although I'm not really using the `component` prop (`button` is fine for our purpose), so this doc is only somewhat relevant.

Reply:
@jonespm, I thought about this some more, and I think you were on to something. The more I looked at this, the more I wasn't crazy about the branching styling, and realized we could rely more on `ButtonProps`. I refactored it here a bit: https://github.com/tl-its-umich-edu/canvas-app-explorer/pull/171/commits/015fc5a2285049a4fcb2452ef897e009019613d9

Reply:
Ah I see, yeah looks good! I think it's easier to read than the other format.